mantidproject/mantid
# Return shared_pointer in Python WorkspaceGroup.getItem()

**Description of work.**

See commit message for in-depth explanation. This has resolved the regression seen in #29939 


**To test:**
- Ensure new unit test passes



*There is no associated issue, this was noticed whilst working on something else*

This does not require release notes because it's obscure enough few people will have run into it outside the development team. Especially as it requires people to only use workspace pointers rather than the ADS



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
